### PR TITLE
Adds a resuable pipeline for acceptance testing

### PR DESCRIPTION
**Pull request by cgranleese-r7**
This PR adds a reusable pipeline to reduce code duplication across our acceptance testing, in various repositories:
- mettle: https://github.com/rapid7/mettle/pull/266
- metasploit-payloads: https://github.com/rapid7/metasploit-payloads/pull/728

## Verification

- [ ] Ensure code changes sane

### CODE REVIEW

File: .github/workflows/meterpreter_acceptance.yml [hunk 1]
```diff
@@ -56,328 +56,8 @@ on:
...
-
-      - name: archive results
-        if: always()
-        uses: actions/upload-artifact@v4
-        with:
-          name: final-report-${{ github.run_id }}
-          path: |
-            ./allure-report
+  build:
+    uses: rapid7/metasploit-framework/.github/workflows/reusable_pipeline.yml@master
```
Comment (cgranleese-r7) [inline .github/workflows/meterpreter_acceptance.yml:60]: This was tested by pointing the job to make use of the new `resuable_pipeline.yml` workflow on my own [branch](https://github.com/cgranleese-r7/metasploit-framework/actions/runs/11384024193) to make sure everything worked as expected. I believe it should now be updated to now point at: 
`rapid7/metasploit-framework/.github/workflows/reusable_pipeline.yml@master`, but these tests will now fail as that file does not yet exist in master. I will call this out in a comment as well.

**Comment by cgranleese-r7** [comment]
Its seems this change has also changed the naming convention for the `Meterpreter Acceptance` job to `.github/workflows/meterpreter_acceptance.yml`:

image

File: .github/workflows/meterpreter_acceptance.yml (hunk 1 above)
Comment (adfoster-r7) [reply .github/workflows/meterpreter_acceptance.yml:60]: Will this mean we're always a step behind when we point to master? i.e. if we put up a PR to change the local `.github/workflows/reusable_pipeline.yml` file - it won't be picked up?

Shouldn't this be running the local definition instead 👀

Comment (cgranleese-r7) [reply .github/workflows/meterpreter_acceptance.yml:60]: Oh, good point. I can update to use the local definition 👍

File: .github/workflows/meterpreter_acceptance.yml [hunk 2]
```diff
@@ -56,328 +56,8 @@ on:
...
-      - name: archive results
-        if: always()
-        uses: actions/upload-artifact@v4
-        with:
-          name: final-report-${{ github.run_id }}
-          path: |
-            ./allure-report
+  build:
+    uses: ./.github/workflows/reusable_pipeline.yml
+    with:
```
Comment (adfoster-r7) [inline .github/workflows/meterpreter_acceptance.yml:61]: We should slot in better names for these now that it's working :+1:

File: .github/workflows/meterpreter_acceptance.yml [hunk 3]
```diff
@@ -56,328 +56,8 @@ on:
...
-
-      - name: archive results
-        if: always()
-        uses: actions/upload-artifact@v4
-        with:
-          name: final-report-${{ github.run_id }}
-          path: |
-            ./allure-report
+  build:
+    uses: ./.github/workflows/reusable_pipeline.yml
```
Comment (adfoster-r7) [inline .github/workflows/meterpreter_acceptance.yml:60]: We should slot in better names for these now that it's working :+1:

Comment (cgranleese-r7) [reply .github/workflows/meterpreter_acceptance.yml:60]: Any suggestions? 👀

Comment (cgranleese-r7) [reply .github/workflows/meterpreter_acceptance.yml]: Went with `.github/workflows/build_payloads.yml`, let me know if you'd prefer something else 👍

File: .github/workflows/shared_meterpreter_acceptance.yml [hunk 4]
```diff
@@ -0,0 +1,343 @@
+name: Build and test payloads
```
Comment (adfoster-r7) [inline .github/workflows/shared_meterpreter_acceptance.yml:1]:
```suggestion
name: Shared Meterpreter Acceptance
```

File: .github/workflows/meterpreter_acceptance.yml [hunk 5]
```diff
@@ -56,328 +56,8 @@ on:
...
-        uses: actions/upload-artifact@v4
-        with:
-          name: final-report-${{ github.run_id }}
-          path: |
-            ./allure-report
+  build:
+    uses: ./.github/workflows/shared_meterpreter_acceptance.yml
+    with:
+      metasploit-payloads_commit: ${{ github.event.inputs.metasploitPayloadsCommit }}
+      mettle_commit: ${{ github.event.inputs.mettleCommit }}
```
Comment (adfoster-r7) [inline .github/workflows/meterpreter_acceptance.yml:63]:
```suggestion
      mettle_commit: ${{ github.event.inputs.mettleCommit }}
      build_metasploit_payloads: contains(github.event.pull_request.labels.*.name, 'payload-testing-branch')
      build_mettle: contains(github.event.pull_request.labels.*.name, 'payload-testing-mettle-branch')
```

And over in mettle we'll just pass in `true` for this to always build

```
build_mettle: true
```

**Review by adfoster-r7 — APPROVED** [review]

**Pull request merged**